src/tests/Directory.Build.targets

```
@@ -527,9 +527,10 @@
     <TrimMode Condition="'$(EnableAggressiveTrimming)' != 'true'">partial</TrimMode>
 
     <IlcToolsPath>$(CoreCLRILCompilerDir)</IlcToolsPath>
-    <IlcToolsPath Condition="'$(TargetArchitecture)' != '$(BuildArchitecture)'">$(CoreCLRCrossILCompilerDir)</IlcToolsPath>
+    <IlcToolsPath Condition="'$(TargetArchitecture)' != '$(BuildArchitecture)' and '$(TargetsAppleMobile)' != 'true'">$(CoreCLRCrossILCompilerDir)</IlcToolsPath>
```

I would think we should be using the CrossILCompilerDir for mobile as well - that's the compiler that is hosted on the current OS/arch. I think the necessary fix for this should be along the lines of:

```suggestion
    <IlcToolsPath Condition="'$(TargetArchitecture)' != '$(BuildArchitecture)' or '$(TargetOS)' != '$(HostOS)''">$(CoreCLRCrossILCompilerDir)</IlcToolsPath>
```

---

This will be updated once the cross-build is resolved.